docs/prepare-image.md

```diff
@@ -0,0 +1,56 @@
+# Bootstraping a snappy device
+
+The `snap prepare-image` command is designed to do the snap-specific
+part of building an image.
+
+## Running it
+
+The `snap prepare-image` command takes a model assertion and a a
+directory (henceforth $ROOT) as intput. It will create `$ROOT/image`
+that will contain the directory layout with content for the image. It
+will also create `$ROOT/gadget` that will contain the unpacked gadget
```

Comment:
In my testing, `snap prepare-image` doesn't create `$ROOT/gadget`; that directory has to already exist:

```
% mkdir /tmp/sroot
% snap prepare-image --channel edge model.assertion /tmp/sroot
40.00 KB / 40.00 KB  100.00 % 4.55 MB/s 
error: unable to create /tmp/sroot/gadget/canonical-pc_8.snap: open /tmp/sroot/gadget/canonical-pc_8.snap: no such file or directory
% ls /tmp/sroot/
% mkdir /tmp/sroot/gadget
% snap prepare-image --channel edge model.assertion /tmp/sroot
40.00 KB / 40.00 KB  100.00 % 4.27 MB/s 
Fetching canonical-pc
40.00 KB / 40.00 KB  100.00 % 850.64 KB/s 
Fetching ubuntu-core
72.45 MB / 72.45 MB  100.00 % 1.25 MB/s 
Fetching canonical-pc-linux
110.62 MB / 110.62 MB  100.00 % 1.02 MB/s 
```